src/Core/DoctorWatson.java

```diff
@@ -0,0 +1,39 @@
+package Core;
+
+import java.util.*;
+
+public class DoctorWatson {
+    private Random random = new Random();
+    private Room currentRoom;
+    private List<String> hints; // All hints loaded from JSON
+    private List<String> remainingHints; // Hints not yet provided
+
+    public DoctorWatson(List<String> hints) {
+        this.hints = new ArrayList<>(hints); // Store all hints
+        this.remainingHints = new ArrayList<>(hints); // Initialize with all hints
+    }
+
+    public Room getCurrentRoom() { return currentRoom; }
+    public void setCurrentRoom(Room room) { currentRoom = room; }
+
+    public void randomMove(Building building) {
+        Map<String, Room> neighbors = currentRoom.getNeighbors();
+        List<Room> allowedRooms = new ArrayList<>(neighbors.values());
+        if (!allowedRooms.isEmpty()) {
+            currentRoom = allowedRooms.get(random.nextInt(allowedRooms.size()));
+        }
+    }
```

Comment:
Move the movement logic to a parent class, since it's the same logic used in both DoctorWatson and Suspect